user/parlib/signal.c

```diff
@@ -74,7 +74,10 @@ static void default_term_handler(int signr, siginfo_t *info, void *ctx)
 static void default_core_handler(int signr, siginfo_t *info, void *ctx)
 {
 	fprintf(stderr, "Segmentation Fault (sorry, no core dump yet)\n");
-	print_user_context((struct user_context*)ctx);
+	if (ctx)
+		print_user_context((struct user_context*)ctx);
+	else
+		fprintf(stderr, "No ctx for %s\n", __FUNCTION__);
```

Comment:
On 2015-10-14 at 16:02 Kevin Klues <notifications@github.com> wrote:
> > @@ -74,7 +74,10 @@ static void default_term_handler(int signr,
> > siginfo_t *info, void *ctx) static void default_core_handler(int
> > signr, siginfo_t *info, void *ctx) {
> >  	fprintf(stderr, "Segmentation Fault (sorry, no core dump
> > yet)\n");
> > -	print_user_context((struct user_context*)ctx);
> > +	if (ctx)
> > +		print_user_context((struct user_context*)ctx);
> > +	else
> > +		fprintf(stderr, "No ctx for %s\n", __FUNCTION__);
> 
> Really?  I feel like we should restrict sys_notify from sending
> signals that generate SIGSEGV and friends?

I had an app that intercepted SIGSEGV, then sent it to itself with
kill.  At the kernel layer, it's just an event.  If we want to change
kill to not allow sigsegv, we can change glibc for that.  Regardless,
we probably ought to be safe in our handler.

Barret

p.s. - i didn't need to CC akaros@ for this email, but i did receive
your email twice, and i'm getting my own emails sent back to me via
github.